src/test/groovy/com/github/rodionovsasha/commentservice/controllers/CommentControllerTest.groovy

```diff
@@ -0,0 +1,73 @@
+package com.github.rodionovsasha.commentservice.controllers
+
+import com.github.rodionovsasha.commentservice.entities.Comment
+import com.github.rodionovsasha.commentservice.entities.Topic
+import com.github.rodionovsasha.commentservice.entities.User
+import com.github.rodionovsasha.commentservice.exceptions.TopicNotFoundException
+import com.github.rodionovsasha.commentservice.services.CommentService
+import org.springframework.http.HttpStatus
+import org.springframework.mock.web.MockHttpServletResponse
+import spock.lang.Specification
+
+import static com.github.rodionovsasha.commentservice.Application.API_BASE_URL
+import static com.github.rodionovsasha.commentservice.controllers.TestUtils.extractJson
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE
+
+class CommentControllerTest extends Specification {
+    final TOPIC_ID = 1
+
+    def service = Mock(CommentService)
+    def controller = new CommentController(service)
+    def user = Mock(User)
+    def topic = Mock(Topic)
+    def comments = [new Comment("Why you little...!", user, topic),
+                    new Comment("Eat My Shorts!", user, topic)]
+
+    def mockMvc = standaloneSetup(controller).setControllerAdvice(new ExceptionHandlerController()).build()
+
+    def "#findByTopic returns comments for topic"() {
+        when:
+        def response = extractJson(findByTopic(TOPIC_ID))
+
+        then:
+        1 * service.findByTopic(TOPIC_ID) >> comments
+        with(response) {
+            it.size == 2
+            id == [0, 0]
+            content == ["Why you little...!", "Eat My Shorts!"]
+            [!archived, !archived]
+            [date instanceof Long, date instanceof Long]
```

Comment:
Is it possible to check like `date*.every {  date instanceof Long }`  and `archived*.every { !archived }` ? (or maybe without s `*` symbol, I might be wrong with exact syntax)

Reply:
:+1: